MarkupsToModel/Logic/vtkSlicerMarkupsToModelLogic.cxx

```diff
@@ -284,16 +286,36 @@ void vtkSlicerMarkupsToModelLogic::UpdateSelectionNode( vtkMRMLMarkupsToModelNod
 //------------------------------------------------------------------------------
 void vtkSlicerMarkupsToModelLogic::UpdateOutputCloseSurfaceModel( vtkMRMLMarkupsToModelNode* markupsToModelModuleNode, vtkPolyData* output )
 {
+  if ( markupsToModelModuleNode == NULL )
+  {
+    vtkErrorMacro( "No markupsToModelModuleNode provided to UpdateOutputCloseSurfaceModel! No operation performed!" );
+    return;
+  }
+
   vtkMRMLMarkupsFiducialNode* markups = markupsToModelModuleNode->GetMarkupsNode();
-  if ( markups == NULL )
+  if ( output == NULL )
   {
+    vtkErrorMacro( "No output poly data provided to UpdateOutputCloseSurfaceModel! No operation performed!" );
     return;
   }
+
+  output->Reset(); // empty the poly data of all existing data
+
+  bool outputContainsData = output->GetNumberOfPoints();
+  if ( outputContainsData )
+  {
+    vtkWarningMacro( "Output poly data provided to UpdateOutputCloseSurfaceModel already contains data. Existing data will be overwritten." );
```

Comment:
It's not normal under the current scheme. The output pointer points to a new poly data object. I think this should be changed, but I think it is part of a larger (different) issue around how updates should be handled.